openslides_votecollector/views.py

```diff
@@ -448,13 +448,25 @@ def post(self, request, poll_id, keypad_id):
             return HttpResponse(_('Vote rejected'))
 
         if vc.voting_mode == 'MotionPoll':
-            conn, created = MotionPollKeypadConnection.objects.get_or_create(poll=poll, keypad=keypad)
-            conn.serial_number = request.POST.get('sn')
+            try:
+                conn = MotionPollKeypadConnection.objects.get(poll=poll, keypad=keypad)
+            except MotionPollKeypadConnection.DoesNotExist:
+                conn = MotionPollKeypadConnection()
+                conn.poll = poll
+                conn.keypad = keypad
+                conn.serial_number = request.POST.get('sn')
+                conn.value = value
+                conn.save()
         else:
-            conn, created = AssignmentPollKeypadConnection.objects.get_or_create(poll=poll, keypad=keypad)
-            conn.serial_number = request.POST.get('sn')
-        conn.value = value
-        conn.save()
+            try:
+                conn = AssignmentPollKeypadConnection.objects.get(poll=poll, keypad=keypad)
+            except AssignmentPollKeypadConnection.DoesNotExist:
+                conn = AssignmentPollKeypadConnection()
+                conn.poll = poll
+                conn.keypad = keypad
+                conn.serial_number = request.POST.get('sn')
```

Comment:
You have to dedent these lines so that the new value for serial_number and value are saved also if then APKC already exists.